.github/workflows/android.yml

```diff
@@ -0,0 +1,36 @@
+name: Build Android APK
+
+on:
+  # Runs on pushes targeting the default branch
+  push:
+    branches:
+      - "main"
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: write
+  pages: write
+  id-token: write
+
+jobs:
+  # Build job
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v4
+      - name: Set up Flutter
+        uses: subosito/flutter-action@v2
+        with:
+          channel: stable
+          flutter-version-file: pubspec.yaml # path to pubspec.yaml
+          cache: true
+      - run: flutter pub get
+      - run: dart run build_runner build
+      - run: flutter test
+      - run: flutter build apk
+      - run: flutter build appbundle
```

Comment:
Why is this neccessary? I already have workflow set up to automatically publish new releases based on tags.